moveit_core/planning_scene/src/planning_scene.cpp

```diff
@@ -492,9 +505,18 @@ void PlanningScene::checkCollision(const collision_detection::CollisionRequest&
   // check collision with the world using the padded version
   getCollisionEnv()->checkRobotCollision(req, res, robot_state, acm);
 
-  // do self-collision checking with the unpadded version of the robot
+  // do self-collision checking with the unpadded version of the robot - unless otherwise specified
   if (!res.collision || (req.contacts && res.contacts.size() < req.max_contacts))
-    getCollisionEnvUnpadded()->checkSelfCollision(req, res, robot_state, acm);
+  {
+    if (use_padding_self_collisions_)
+    {
+      getCollisionEnv()->checkSelfCollision(req, res, robot_state, acm);
```

Comment:
Why not just replace one line so that
 ```diff
- 497 getCollisionEnvUnpadded()->checkSelfCollision(req, req, robot_state, acm);
+ 497 getCollisionEnv()->checkSelfCollision(req, req, robot_state, acm);
```

Reply:
I think it is just setting ROS param to enable or not to enable in order not to change the default behavior.

Reply:
However, I don't think this parameter implementation is very good, and if there is a better way, I would like to change the place where the parameter is loaded

Reply:
I think using the unpadded collision check is intentional. 
https://github.com/sbgisen/moveit/blob/def489bb60a4aa48e6b0fe7c60ddee9939548d30/moveit_core/planning_scene/src/planning_scene.cpp#L492-L497

It seems that for self-collision check (checking collision within links), MoveIt deems it unnecessary to use padding parameter.

Reply:
Padding is required for self-collision check.

Reply:
I'm thinking we should follow [this](https://github.com/ros-planning/moveit/pull/2840#issuecomment-1196542894) comment about the names of the parameters.

Reply:
> I'm thinking we should follow [this](https://github.com/ros-planning/moveit/pull/2840#issuecomment-1196542894) comment about the names of the parameters.

Parameter name is ok. Just wondering if we ever need a flag to disable/enable self-collision check with padding. I can't think of a scenario where we want to pad the robot arm with the environment but disable padding on self-collision.

Reply:
I git it.